### PR TITLE
Remove sub-modules under package/

### PR DESCRIPTION
**Pull request by zhztheplayer**
This is to

1. simplify codes
2. add more flexibility, for example to package more than one backends into the bundle jar, if we support that in future

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -114,13 +116,16 @@
...
     <profile>
-        <id>spark-3.2</id>
-        <properties>
-          <scala.version>${spark32.scala}</scala.version>
-          <spark.version>${spark32.version}</spark.version>
-          <delta.version>${delta1.version}</delta.version>
-          <sparkbundle.version>${spark32bundle.version}</sparkbundle.version>
-        </properties>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```
Comment (jinchengchenghh) [inline pom.xml:120]: Why need this one?

Comment (zhztheplayer) [reply pom.xml:120]: We either remove all `<activeByDefault>true</activeByDefault>` under `<id>spark-3.2</id>`, either add to them all. Maybe we just remove all of them?

**Comment by zhztheplayer** [comment]
> Please also change the jenkins.

Yes. and as well as arrow's GHA CI. Please help me approve this so I can make the upgrade in short time.

File: pom.xml (hunk 1 above)
Comment (jinchengchenghh) [reply pom.xml:120]: In the old logical, it will activate -Pspark at default, if we remove -Pspark, I think we can add to them all.

**Review by jinchengchenghh — APPROVED** [review]

**Pull request merged**